telegram/ext/filters.py

```diff
@@ -865,73 +865,83 @@ class user(BaseFilter):
             user_ids(set(:obj:`int`), optional): Which user ID(s) to allow through.
             usernames(set(:obj:`str`), optional): Which username(s) (without leading '@') to allow
                 through.
+            allow_empty(:obj:`bool`, optional): Whether updates should be processed, if no user
+                is specified in :attr:`user_ids` and :attr:`usernames`.
         Args:
             user_id(:obj:`int` | List[:obj:`int`], optional): Which user ID(s) to allow
                 through.
             username(:obj:`str` | List[:obj:`str`], optional): Which username(s) to allow
                 through. If username starts with '@' symbol, it will be ignored.
+            allow_empty(:obj:`bool`, optional): Whether updates should be processed, if no user
+                is specified in :attr:`user_ids` and :attr:`usernames`. Defaults to :obj:`False`
 
         Raises:
             ValueError: If chat_id and username are both present, or neither is.
 
         """
 
-        def __init__(self, user_id=None, username=None):
-            if (user_id is None) == (username is None):
-                raise ValueError('One and only one of user_id or username must be used')
+        def __init__(self, user_id=None, username=None, allow_empty=False):
```

Comment:
Personally, I don't think this would be a good fit for defaults, copying from dev chat:

> Right now those are only used for arguments for a) bot methods, which b) appear in multiple different methods c) are well established

But maybe I change my mint for a next iteration